validationTemplate/src/validationWorkflowGenerator.py

```diff
@@ -0,0 +1,413 @@
+"""
+Created on July 28th 2021
+@author: garrm
+Template interface for the RAVEN Validation template
+"""
+from __future__ import print_function, unicode_literals
+import os
+import shutil
+import sys
+import inspect
+import configparser
+import logging
+import subprocess
+from datetime import datetime
+from collections import OrderedDict
+from typing import Dict
+from _utils import get_raven_loc
+import _utils as hutils
+from ModelValidationTemplate.ValidationTemplate import ValidationTemplate
+from ModelCalibrationTemplate.CalibrationTemplate import CalibrationTemplate
+from ROMTemplate.ROMTemplate import ROMTemplate
+
+# Authorized values dictionary
+# TODO Change the way input testing is done ask P. Talbot about how they do it in RAVEN
+authorized = {'type': ['calibration', 'validation'],
+                'metrics': {'DTW':'DTW', 
+                            'mean_squared_error':['SKL', 'regression'],
+                            'mean_absolute_error':['SKL', 'regression'],
+                            'euclidean':['SKL', 'paired_distance']}, 
+                'distributions' : {'Uniform':['lowerBound','upperBound'],
+                                'Normal':['mean', 'sigma']},
+                'metrics_sync' :{'DTW':False, 
+                                'mean_squared_error':True, 
+                                'mean_absolute_error':True, 
+                                'euclidean':True}}
+roms = {'DMD': {'required':['dmdType', 'rankSVD'],
+                'optional':['optimized', 'exactModes','rankTLSQ','energyRankSVD']}, 
+        'PolyExponential':{'required': ['numberExpTerms', 'tol', 'max_iter'], 
+                            'optional':['coeffREgressor']}, 
+        'SyntheticHistory':{'required': None, 
+                            'optional': ['fourier', 'wavelet']} }
+samplers = ['MonteCarlo', 'Stratified']
+authorized['roms'] = roms
+authorized['samplers'] = samplers
+
+
+
+
+def readInputFile(inputFilePath, logger):
+  """ 
+  Read input file 
+      @ In, inputFilePath, string, relative path to the input file
```

Comment:
it looks like many tabs are 4 spaces in this file instead of the 2 required by HYBRID SQA